clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -634,6 +639,55 @@ public void testDeleteAcls() throws Exception {
         }
     }
 
+    @Test
+    public void testElectPreferredLeaders()  throws Exception {
+        TopicPartition topic1 = new TopicPartition("topic", 0);
+        TopicPartition topic2 = new TopicPartition("topic", 2);
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+
+            // Test a call where one partition has an error.
+            ApiError value = ApiError.fromThrowable(new ClusterAuthorizationException(null));
+            ElectPreferredLeadersResponseData responseData = new ElectPreferredLeadersResponseData();
+            ReplicaElectionResult r = new ReplicaElectionResult().setTopic(topic1.topic());
+            r.partitionResult().add(new PartitionResult()
+                    .setPartitionId(topic1.partition())
+                    .setErrorCode(ApiError.NONE.error().code())
+                    .setErrorMessage(ApiError.NONE.message()));
+            r.partitionResult().add(new PartitionResult()
+                    .setPartitionId(topic2.partition())
+                    .setErrorCode(value.error().code())
+                    .setErrorMessage(value.message()));
+            responseData.replicaElectionResults().add(r);
+            env.kafkaClient().prepareResponse(new ElectPreferredLeadersResponse(responseData));
+            ElectPreferredLeadersResult results = env.adminClient().electPreferredLeaders(asList(topic1, topic2));
+            results.partitionResult(topic1).get();
+            TestUtils.assertFutureError(results.partitionResult(topic2), ClusterAuthorizationException.class);
+            TestUtils.assertFutureError(results.all(), ClusterAuthorizationException.class);
```

Comment:
Perhaps we can further assert that calling get on topic1 has no error?

Reply:
We do that on line 664, unless I misunderstood what you mean?